crates/mun_gc/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+mod mark_sweep;
+mod ptr;
+mod root_ptr;
+
+pub use mark_sweep::MarkSweep;
+pub use ptr::{GcPtr, HasIndirectionPtr, RawGcPtr};
+pub use root_ptr::GcRootPtr;
+use std::marker::PhantomData;
+
+/// Contains stats about the current state of a GC implementation
+#[derive(Debug, Clone, Default)]
+pub struct Stats {
+    pub allocated_memory: usize,
+}
+
+/// A trait used by the `GcRuntime` to identify an object type.
+pub trait Type: Send + Sync {
+    type Trace: Iterator<Item = GcPtr>;
+
+    /// Returns the size of an object of this type (in bytes).
+    fn size(&self) -> usize;
```

Comment:
I just realised, we could have the `Type` trait return a `Layout`, since that encapsulates both `size` and `alignment`. It simplifies this API and is conform the `Alloc` trait, so makes usage easier too. WDYT?

Reply:
Good idea, I changed it.